### PR TITLE
SAMBRO: shorten alert url

### PR DESCRIPTION
**Pull request by biplovbhandari**

### CODE REVIEW

File: controllers/cap.py [hunk 1]
```diff
@@ -1407,6 +1407,15 @@ def notify_approver():
                 subject = "%s: Alert Approval Required" % settings.get_system_name_short()
                 url = "%s%s" % (settings.get_base_public_url(),
                                 URL(c="cap", f="alert", args=[alert_id, "review"]))
+                try:
+                    from pyshorteners import Shortener
```
Comment (flavour) [inline controllers/cap.py]: Should be added as an optional dependency to requirements.txt

File: controllers/cap.py [hunk 2]
```diff
@@ -1407,6 +1407,15 @@ def notify_approver():
...
+                    try:
+                        url = s3_str(Shortener('Tinyurl', timeout=3).short(url))
+                    except:
+                        pass
+                except ImportError:
```
Comment (flavour) [inline controllers/cap.py]: Better to do as:
try:
    import
except Import Error
    pass
else:
    try:
        url = ...

Comment (flavour) [reply controllers/cap.py]: & ideally 2nd except would be more specific...why would it fail? Network issues?

File: modules/templates/SAMBRO/config.py [hunk 3]
```diff
@@ -450,21 +450,28 @@ def onapprove(record):
...
                         current.log.debug("tweepy module needed for sending tweets")
                     else:
+                        url = "%s/%s" % (arow["cap_info.web"], "profile")
+                        try:
+                            from pyshorteners import Shortener
+                            try:
+                                url = Shortener('Tinyurl', timeout=3).short(url)
+                            except:
+                                pass
+                        except ImportError:
```
Comment (flavour) [inline modules/templates/SAMBRO/config.py]: Sam try/except/else

File: controllers/cap.py (hunk 2 above)
Comment (biplovbhandari) [reply controllers/cap.py]: the exception is ```requests.exceptions.ReadTimeout```. Just do not want to import requests, which would be unused. I would make changes with the structure you mentioned

**Pull request merged**